### PR TITLE
Make sure that supporting external parameters will recompile a PyTEAL program

### PR DESCRIPTION
**Pull request by amityadav0**
- make test that this works as expected.
   - Test both stateful and stateless smart contracts

### CODE REVIEW

File: packages/algob/test/builtin-tasks/compile.ts [hunk 1]
```diff
@@ -125,3 +125,71 @@ describe("Compile task", () => {
...
+  it("On second run, shouln't recompile because No external parameters passed", async () => {
+    await op.resetAndCompile(false);
+
+    // timestamp didn't change because file is not recompiled
+    assert.equal(op.timestamp, 2);
+    assert.lengthOf(op.compiledFiles, 0);
+    assert.lengthOf(op.writtenFiles, 0);
+  });
+
+  it("On third run, should compile on third run because external parameters passed", async () => {
```
Comment (robert-zaremba) [inline packages/algob/test/builtin-tasks/compile.ts:173]: will the last 2 unit tests work if the first one will fail? If not then we can put all of them in a single test (`it(...)`)

File: packages/algob/test/fixture-projects/support-external-parameters/algob.config.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  token: 'some-fake-token',
+  wallets: [
+    {
+      name: 'some-fake-wallet',
+      password: '',
+      accounts: [
+        { name: 'name-1', address: 'some-addr' }]
+    }
+  ]
+}
```
Comment (robert-zaremba) [inline packages/algob/test/fixture-projects/support-external-parameters/algob.config.js:27]: do we need `kmdCfg` for this test project?

File: packages/algob/test/fixture-projects/support-external-parameters/algob.config.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      token: 'somefaketoken',
+      accounts: accounts
+    },
+    kmdNet: {
+      host: 'http://127.0.0.1',
+      port: 8080,
+      token: 'somefaketoken',
+      accounts: accounts,
+      kmdCfg: kmdCfg
+    }
```
Comment (robert-zaremba) [inline packages/algob/test/fixture-projects/support-external-parameters/algob.config.js:52]: I think one network is enough here.

File: packages/algob/test/fixture-projects/support-external-parameters/algob.config.js (hunk 2 above)
Comment (amityadav0) [reply packages/algob/test/fixture-projects/support-external-parameters/algob.config.js:27]: No, cleaning config

File: packages/algob/test/fixture-projects/support-external-parameters/algob.config.js (hunk 3 above)
Comment (amityadav0) [reply packages/algob/test/fixture-projects/support-external-parameters/algob.config.js:52]: ok

File: packages/algob/test/builtin-tasks/compile.ts (hunk 1 above)
Comment (amityadav0) [reply packages/algob/test/builtin-tasks/compile.ts:173]: ok

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**